kyo-core/shared/src/main/scala/kyo/Channel.scala

```diff
@@ -105,16 +105,18 @@ object Channel:
         def takeExactly(n: Int)(using Frame): Chunk[A] < (Abort[Closed] & Async) =
             if n <= 0 then Chunk.empty
             else
-                Loop(Chunk.empty[A]): lastChunk =>
-                    val nextN = n - lastChunk.size
+                Loop(Chunk.empty[A], 0): (lastChunk, lastSize) =>
+                    val nextN = n - lastSize
                     Channel.drainUpTo(self)(nextN).map: chunk =>
                         val chunk1 = lastChunk.concat(chunk)
+                        val size1  = chunk1.size
```

Comment:
`Chunk.size` returns in constant time so it's best to get it on each iteration instead of capturing `size1`, which requires an additional field in the generated function